### ISSUE TITLE
🐛 Output from git diff -z is cut off when using delta

### ISSUE THREAD

**Issue by paleite** [dandavison/delta#863]
Thanks for a great job with delta overall!

I ran into this bug recently:

When setting delta as my pager the output from git gets cut off when using -z if the output is long enough (somewhere above 500 characters, I think). See output below. Note how the last output is piped and working perfectly. I guess this is to be expected because of how pagers are enabled, but I'm not certain.

Using `delta`, running `git diff  --name-only -z HEAD~1`:
```
08ec7738-9ad9-446d-82bb-fa279ee0a7db.txt^@251de719-ef8c-46b1-8695-25ba0c7c5e13.txt^@26ac21c5-4213-4938-b918-0fb980c5751c.txt^@57e8188b-219b-4bc5-affd-2192d5d3726b.txt^@5b754c46-5b62-402c-aef5-c7fe8896f7ea.txt^@75da4f07-0ea5-4c17-a088-70250c6e0955.txt^@8a804894-55cf-4a47-b76f-3a11819ff513.txt^@93c8fbbe-6c30-40a9-815f-07e0e26d0253.txt^@b49774ee-c6da-4c33-ba2e-d4ce3312008a.txt^@bdf976aa-6d2e-4f17-81a3-88eda585fc3b.txt^@c3b4bd8e-3e5d-43f7-af7c-71a61f059a57.txt^@cf99063d-6ff0-48c2-b507-49b686937291.txt^@d9435462-9e2a-49fb-b327-95051806e500.txt^@e1ca99eb-adbe-413d-9e7d-0bb0cf96be51.txt^@e1f085a5-e0→
```

Without `delta`, running `git diff  --name-only -z HEAD~1`:
```
08ec7738-9ad9-446d-82bb-fa279ee0a7db.txt^@251de719-ef8c-46b1-8695-25ba0c7c5e13.txt^@26ac21c5-4213-4938-b918-0fb980c5751c.txt^@57e8188b-219b-4bc5-affd-2192d5d3726b.txt^@5b754c46-5b62-402c-aef5-c7fe8896f7ea.txt^@75da4f07-0ea5-4c17-a088-70250c6e0955.txt^@8a804894-55cf-4a47-b76f-3a11819ff513.txt^@93c8fbbe-6c30-40a9-815f-07e0e26d0253.txt^@b49774ee-c6da-4c33-ba2e-d4ce3312008a.txt^@bdf976aa-6d2e-4f17-81a3-88eda585fc3b.txt^@c3b4bd8e-3e5d-43f7-af7c-71a61f059a57.txt^@cf99063d-6ff0-48c2-b507-49b686937291.txt^@d9435462-9e2a-49fb-b327-95051806e500.txt^@e1ca99eb-adbe-413d-9e7d-0bb0cf96be51.txt^@e1f085a5-e094-4a6b-aab1-bd6362a955af.txt^@
```

With or without `delta`, running `git diff  --name-only -z HEAD~1 | xargs -0 -n 1 echo`
```
08ec7738-9ad9-446d-82bb-fa279ee0a7db.txt
251de719-ef8c-46b1-8695-25ba0c7c5e13.txt
26ac21c5-4213-4938-b918-0fb980c5751c.txt
57e8188b-219b-4bc5-affd-2192d5d3726b.txt
5b754c46-5b62-402c-aef5-c7fe8896f7ea.txt
75da4f07-0ea5-4c17-a088-70250c6e0955.txt
8a804894-55cf-4a47-b76f-3a11819ff513.txt
93c8fbbe-6c30-40a9-815f-07e0e26d0253.txt
b49774ee-c6da-4c33-ba2e-d4ce3312008a.txt
bdf976aa-6d2e-4f17-81a3-88eda585fc3b.txt
c3b4bd8e-3e5d-43f7-af7c-71a61f059a57.txt
cf99063d-6ff0-48c2-b507-49b686937291.txt
d9435462-9e2a-49fb-b327-95051806e500.txt
e1ca99eb-adbe-413d-9e7d-0bb0cf96be51.txt
e1f085a5-e094-4a6b-aab1-bd6362a955af.txt
```

**Comment by dandavison** [comment]
Hi @paleite, thanks.

Delta truncates long lines in diff output by default: see `max-line-length`. (This prevents things like trying to syntax extremely long lines of minified javascript). You can disable truncation for this command only with

```
git -c delta.max-line-length=0 diff --name-only -z HEAD~1
```

or, as you note, redirecting output to file prevents git from invoking its pager at all (and will probably be natural for this command, right?)

Technically I think you're right that this is a bug in that this is a special form of output that the user almost certainly wants untruncated, so `max-line-length` probably should not be honored. We could perhaps detect this by inspecting the calling process command line, which is something delta does nowadays to do things like determine the language for `git show $commit:$file.lang` and `git blame` output, switching to expect grep output, and supporting `--word-diff` / `--color-words` output, etc. I'd be happy to consider a PR; do you think its worth it, or will users in practice always redirect this output?

**Comment by paleite** [comment]
Yeah, good point! The reason I came across this bug was that I was trying to make sense of some output that I compared to some expected output and noticed that it was incomplete. `delta` was not the issue in that case, but I noticed the bug.
After your reply, I'm convinced it's probably not an issue worth fixing since, as you said, users will in practice always redirect the output.
Thanks for your time!
